foundation/foundation-upgrade/src/main/java/com/boozallen/aissemble/upgrade/migration/HelmTemplatesMigration.java

```diff
@@ -0,0 +1,92 @@
+package com.boozallen.aissemble.upgrade.migration;
+
+/*-
+ * #%L
+ * aiSSEMBLE::Foundation::Upgrade
+ * %%
+ * Copyright (C) 2021 Booz Allen
+ * %%
+ * This software package is licensed under the Booz Allen Public License. All Rights Reserved.
+ * #L%
+ */
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.technologybrewery.baton.BatonException;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.FileVisitResult;
+import java.nio.file.attribute.BasicFileAttributes;
+
+
+/**
+ * Migration to update the aissemble version used in the helmfile
+ */
+public class HelmTemplatesMigration extends AbstractAissembleMigration {
+
+    private static final Logger logger = LoggerFactory.getLogger(HelmTemplatesMigration.class);
+    private static final String ARGOCD_REGEX = "argocd";
+
+
+    /**
+     * Determines if migration should be executed. Will execute if there are files in the *-deploy/templates folder
+     *
+     * @param file values file to check
+     * @return true if it needs to be updated
+     */
+    @Override
+    protected boolean shouldExecuteOnFile(File file) {
+        boolean shouldExecute = true;
+        try (BufferedReader valuesFile = new BufferedReader((new FileReader(file)))) {
```

Comment:
More of an FYSA since I know some people think the pattern you have here is more readable, but I personally prefer:
```suggestion
        try (BufferedReader valuesFile = Files.newBufferedReader(file.toPath)) {
```

Reply:
Much cleaner

- [ ] Tidy code